core/src/test/java/com/linecorp/armeria/client/TestReconfigurableServer.java

```diff
@@ -0,0 +1,225 @@
+        assertThat(response11.status()).isEqualTo(HttpStatus.OK);
+        assertThat(response11.contentUtf8()).isEqualTo("Hello, WORLD");
+
+        Thread.sleep(1000); // Some time goes by now want to reconfigure new server
+        System.out.println("Configuring new server");
+        server.reconfigure(serverBuilder  -> {
+            // Need to reconfigure ssl context.
+            serverBuilder.tlsSelfSigned();
+            // Replace the entire routes with the following two services.
+            serverBuilder.service("/test2", (ctx, req) -> HttpResponse.of("Hello, world!"));
+
+            serverBuilder.service("/test2/{name}",
+                    (ctx, req) ->
+                            HttpResponse.of("Hello, " + ctx.pathParam("name").toUpperCase()));
+        });
+
+        // retry calling the original service. Should return 404.
+        final AggregatedHttpResponse response1 = originalClient.get("/test11/world").aggregate().join();
+        assertThat(response1.status()).isEqualTo(HttpStatus.NOT_FOUND);
+
+        final AggregatedHttpResponse res = originalClient.get("/test2").aggregate().join();
+        assertThat(res.status()).isEqualTo(HttpStatus.OK);
+        assertThat(res.contentUtf8()).isEqualTo("Hello, world!");
+
+        //Open new connection to check if the newly reconfigured server is now operational
+        // and old service endpoints are no longer visible.
+        final WebClient client2 = WebClient.builder("https://localhost:" + server.activeLocalPort())
+                .factory(ClientFactory.insecure())
+                .build();
+
+        final AggregatedHttpResponse response2 = client2.get("/test2/world")
+                .aggregate()
+                .get();
+
+        assertThat(response2.status()).isEqualTo(HttpStatus.OK);
+        assertThat(response2.contentUtf8()).isEqualTo("Hello, WORLD");
+
+        final AggregatedHttpResponse response22 = client2.get("/test2")
+                .aggregate()
+                .get();
+
+        assertThat(response22.status()).isEqualTo(HttpStatus.OK);
+        assertThat(response22.contentUtf8()).isEqualTo("Hello, world!");
+
+        final AggregatedHttpResponse failedResponse = client2.get("/test1").aggregate().get();
+        assertThat(failedResponse.status()).isEqualTo(HttpStatus.NOT_FOUND);
+
+        // Tests that original service configurations are no longer active when you open a new connection
+        // with the server.
+        final WebClient client3 = WebClient.builder("https://localhost:" + server.activeLocalPort())
+                .factory(ClientFactory.insecure())
+                .build();
+
+        final AggregatedHttpResponse response3 = client3.get("/test1")
+                .aggregate()
+                .get();
+        assertThat(response3.status()).isEqualTo(HttpStatus.NOT_FOUND);
+
+        final AggregatedHttpResponse response4 = client3.get("/test1/world")
+                .aggregate()
```

Comment:
Could you also fix the code style of this class?
https://armeria.dev/community/developer-guide/#setting-up-your-ide

Reply:
Done.